### ISSUE TITLE
Treatment of residues with alternative conformation or multiple atoms of the same name in Ramachandran Analysis

### ISSUE THREAD

**Issue by ozyo** [MDAnalysis/mdanalysis#3812]
# Is your feature request related to a problem?

If a residue has multiple atoms with the same name, they are silently skipped over without any warnings. One can provide a selection that ignores alternative conformations, however, the function extracts residues before and after which might contain alternative conformations.

The ideal way of course is to always provide a cleaned-up coordinate system for the calculations. But for the cases where users don't at least a warning should be printed out.
# Describe the solution you'd like

* A warning should be printed out after residues are filtered based according to the atom names (see linked lines). This warning is not just relevant for alternative conformations.

* A really nice to have would be a way to ignore the alternative conformations in the previous and next residue selections within the Ramachandran function.

# Describe alternatives you've considered
To make sure there aren't any alternative conformations I have used a workaround to clean the structure with MDAnalysis, which keeps the input PDB file the same but ensures that the calculations run as expected.

```
# gets rid of multiple frames and any alternative conformation that is not A.
universe = mda.Merge(universe.select_atoms("not altloc B C D E F G"))
```

# Additional context
The lines that filters the residues silently :
https://github.com/MDAnalysis/mdanalysis/blob/cb6e4f50ad93aa7a32db7bc9c6d62c36ffaa1649/package/MDAnalysis/analysis/dihedrals.py#L424-L428
